src/dag/topological-sort.js

```diff
@@ -59,11 +59,13 @@ function forEachNodeInTopologicalOrder(graph, callback) {
   }
 
   while (queue.size !== 0) {
-    const [node, attr] = queue.shift();
+    const [node, attr, gen] = queue.shift();
 
-    callback(node, attr);
+    callback(node, attr, gen);
 
-    graph.forEachOutNeighbor(node, neighborCallback);
+    graph.forEachOutNeighbor(node, (node, attr) => {
```

Comment:
Tu viens de faire ce qu'on avait dit qu'il ne fallait pas faire :). A savoir créer une fonction à chaque itération de la boucle `while`. Ca et tu shadow la variable `node` de manière assez violente là.

Reply:
Il te faut une variable d'état `currentGeneration` que tu set avant d'itérer sur les voisins, que tu utilises dans `neighborCallback` qui n'a plus besoin de l'argument `gen` et l'affaire est dans le sac.